packages/core/src/app/customer/Customer.tsx

```diff
@@ -393,195 +208,167 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps & Ana
         }
 
-        onChangeViewType(CustomerViewType.Login);
-    };
+        onChangeViewType(CustomerViewType.Guest);
+    }, [customerData.actions.clearError, customerData.data.signInError, onChangeViewType]);
 
-    private handleCancelSignIn: () => void = () => {
-        const { clearError, onChangeViewType = noop, signInError } = this.props;
+    const showCreateAccount = useCallback(() => {
+        onChangeViewType(CustomerViewType.CreateAccount);
+    }, [onChangeViewType]);
 
-        if (signInError) {
-            clearError(signInError);
+    const handleSendLoginEmail = useCallback(async (values: EmailLoginFormValues) => {
+        try {
+            await customerData.actions.sendLoginEmail(values);
+        } catch {
+            // Need to write catch block since one test covers the case when `sendLoginEmail` fails
+        } finally {
+            setState(prevState => ({ ...prevState, hasRequestedLoginEmail: true }));
         }
+    }, [customerData.actions.sendLoginEmail]);
 
-        onChangeViewType(CustomerViewType.Guest);
-    };
+    const handleEmailLoginClicked = useCallback(async () => {
+        try {
+            if (viewType !== CustomerViewType.Login && state.draftEmail) {
+                await handleSendLoginEmail({ email: state.draftEmail });
+            }
+        } finally {
+            setState(prevState => ({ ...prevState, isEmailLoginFormOpen: true }));
+        }
+    }, [viewType, state.draftEmail, handleSendLoginEmail]);
 
-    private handleChangeEmail: (email: string) => void = (email) => {
-        const { analyticsTracker } = this.props;
+    const closeEmailLoginFormForm = useCallback(() => {
+        setState(prevState => ({
+            ...prevState,
+            isEmailLoginFormOpen: false,
+            hasRequestedLoginEmail: false,
+        }));
+    }, []);
 
-        this.draftEmail = email;
-        analyticsTracker.customerEmailEntry(email);
-    };
+    // Render methods converted to useCallback
+    const renderGuestForm = useCallback(() => {
+        return (
+            <GuestFormContainer
+                email={state.draftEmail || customerData.data.email}
+                handleChangeEmail={handleChangeEmail}
+                handleContinueAsGuest={handleContinueAsGuest}
+                handleShowLogin={handleShowLogin}
+                isFloatingLabelEnabled={customerData.data.isFloatingLabelEnabled}
+                isSubscribed={isSubscribed}
+                isWalletButtonsOnTop={isWalletButtonsOnTop}
+                onUnhandledError={onUnhandledError}
+                onWalletButtonClick={onWalletButtonClick}
+                step={step}
+            />
+        );
+    }, [state.draftEmail, customerData.data.email, customerData.data.isFloatingLabelEnabled, handleChangeEmail, handleContinueAsGuest, handleShowLogin, isSubscribed, isWalletButtonsOnTop, onUnhandledError, onWalletButtonClick, step]);
```

Comment:
Can we place the component directly into the `return`?

Reply:
Do it in a follow up PR.